src/Core/Checkout/Document/Renderer/ZugferdEmbeddedRenderer.php

```diff
@@ -0,0 +1,133 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Document\Renderer;
+
+use horstoeko\zugferd\ZugferdDocumentPdfMerger;
+use Shopware\Core\Checkout\Document\DocumentException;
+use Shopware\Core\Checkout\Document\Struct\DocumentGenerateOperation;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Feature;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Plugin\Exception\DecorationPatternException;
+
+#[Package('after-sales')]
+class ZugferdEmbeddedRenderer extends AbstractDocumentRenderer
+{
+    public const TYPE = 'zugferd_embedded_invoice';
+
+    /**
+     * @internal
+     */
+    public function __construct(
+        protected AbstractDocumentRenderer $invoiceRenderer,
+        protected AbstractDocumentRenderer $electronicRenderer,
+        protected string $shopwareVersion
+    ) {
+    }
+
+    public function supports(): string
+    {
+        return self::TYPE;
+    }
+
+    public function getDecorated(): AbstractDocumentRenderer
+    {
+        throw new DecorationPatternException(self::class);
+    }
+
+    public function render(array $operations, Context $context, DocumentRendererConfig $rendererConfig): RendererResult
+    {
+        $invoice = $this->invoiceRenderer->render($operations, $context, $rendererConfig);
+
+        if (!Feature::isActive('v6.7.0.0')) {
+            return $invoice;
+        }
+
+        return $this->embedXMLIntoPDF($operations, $context, $rendererConfig, $invoice);
+    }
+
+    /**
+     * @deprecated tag:v6.7.0 - will be removed without replacement
+     */
+    public function finalize(DocumentGenerateOperation $operation, Context $context, DocumentRendererConfig $rendererConfig, RendererResult $result): void
+    {
+        Feature::triggerDeprecationOrThrow('v6.7.0.0', 'Method will be removed without replacement');
+
+        $invoiceResult = new RendererResult();
+        $successDocument = $result->getOrderSuccess($operation->getOrderId());
+        if (!$successDocument) {
+            throw DocumentException::generationError('Success document not found');
+        }
+
+        $invoiceResult->addSuccess($operation->getOrderId(), $successDocument);
+
+        $embeddedResult = $this->embedXMLIntoPDF([$operation->getOrderId() => $operation], $context, $rendererConfig, $result);
+
+        $orderError = $embeddedResult->getOrderError($operation->getOrderId());
+        if ($orderError) {
+            throw $orderError;
+        }
+    }
+
+    /**
+     * @param DocumentGenerateOperation[] $operations
```

Comment:
do you mean 
```suggestion
     * @param list<DocumentGenerateOperation> $operations
```
?
It is more precise. `DocumentGenerateOperation[]` can also be `array<string, DocumentGenerateOperation>` or `array<int, DocumentGenerateOperation>`

Reply:
looking further down, it is probably 

```suggestion
     * @param array<string, DocumentGenerateOperation> $operations
```

Reply:
Do we have to change it here too? It's the same array
1. https://github.com/shopware/shopware/blob/trunk/src/Core/Checkout/Document/Renderer/DocumentRendererRegistry.php#L23
2. https://github.com/shopware/shopware/blob/trunk/src/Core/Checkout/Document/Renderer/AbstractDocumentRenderer.php#L20

Reply:
would be nice, if they also could be adjusted. the "generic" annotation type is more precise about the key the array has

Reply:
@mitelg PHPStan will throw an error `src/Core/Checkout/Document/DocumentGeneratorController.php`
```
Parameter #2 $operations of method Shopware\Core\Checkout\Document\Service\DocumentGenerator::generate() expects array<string, Shopware\Core\Checkout\Document\Struct\DocumentGenerateOperation>,  
         array<int|string, Shopware\Core\Checkout\Document\Struct\DocumentGenerateOperation> given.                                                                                                         
         🪪  argument.type
```
To cast will be a BC, right? 
`$operations[(string) $operation['orderId']] = new DocumentGenerateOperation()`
```php
/** @var string $orderId */
$orderId = $operation['orderId'];
$operations[$orderId] = new DocumentGenerateOperation(
```

Reply:
it would not be a break. it is safe to cast this to string, also it is probably not necessary, as an orderId should always be an UUID string. 

but PHPStan can of course not tell, what the array contains which comes back from the serializer. so a cast is okay for me